### ISSUE TITLE
Complex SVD path: _fix_svd_signs breaks the reconstruction, and the adjoint uses the wrong cotangent convention

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#751]
Found while fixing #750. Two further defects in the same SVD path, both **complex-only** — which is why the real-valued test suite never saw them. Verified against finite differences taken on tenax's own gauge-fixed forward (gauge-free by construction), with the FD harness itself calibrated against closed-form gradients and against `jnp.linalg.svd`'s VJP.

## 751a — `_fix_svd_signs` destroys the factorisation for complex input

`src/tenax/algorithms/_ad_primitives.py`

```python
U  = U  * jnp.conj(phases)[None, :]
Vh = Vh * jnp.conj(phases)[:, None]   # <-- same factor, not the inverse
```

Column `j` of `U` and row `j` of `Vh` both feed term `j` of `U diag(s) Vh`, so scaling both by `conj(p_j)` multiplies that term by `conj(p_j)**2`. That is 1 only when `p_j = ±1` (the real case). For a genuine complex phase the SVD stops being an SVD:

| input | `‖U s Vh − M‖` raw | after `_fix_svd_signs` |
|---|---|---|
| real 6×6 | 3.55e-15 | 3.55e-15 |
| **complex 6×6** | 8.38e-15 | **1.07e+01** |

`‖M‖ ≈ 10`, so that is 100% error. `U` and `Vh` stay individually isometric, which is why orthonormality-based checks pass.

Fix: pair the factors so they cancel — `Vh = Vh * phases[:, None]`. The gauge condition (max-|U| entry real-positive) is unaffected.

## 751b — the adjoint is stated in the wrong cotangent convention, and drops the complex phase term

Two parts:

1. **Convention.** The kernel assembles the textbook adjoint (Townsend 2016; Wan & Zhang 2019), whose cotangents are `dL/dX`. JAX's cotangents for a real-valued loss of a complex input are the conjugate (`grad` returns `dL/dRe − i dL/dIm`). Isolating term 1 with `loss = sum(s)` (so `dU = dVh = 0`) shows it exactly: as-returned `rel=1.57e+00, cos=−0.228`; conjugated `rel=8.6e-09, cos=1.000000000`.
2. **Missing term.** The complex-only Wan & Zhang phase term `½ U S⁻¹ (Lᴴ − L) Vᴴ`, with `L = I∘(Vᴴ dV)`, was absent. `Lᴴ − L` is a purely imaginary diagonal, so it vanishes identically on real input.

Measured on a gauge-invariant complex loss `Re⟨W, U s² Vh⟩`, 6×6, after the #750 fix:

| state | rel | cos |
|---|---|---|
| #750 fix only | 2.00e+00 | −0.057 |
| \+ conjugation convention | 1.37e-01 | 0.991 |
| \+ phase term | **7.8e-09** | **1.000000000** |

## Known remaining limitation (not a defect)

For a **phase-dependent** complex loss — one reading `U` or `Vh` entrywise — agreement stops at `rel ≈ 0.15`. `_fix_svd_signs` pins a continuous phase for complex input, and the adjoint does not differentiate through that gauge fixing. (In the real case the gauge is a discrete ±1, locally constant, so it contributes nothing and phase-dependent losses are exact.) Every physical loss — energies, observables, CTM projectors composed into a fixed point — is gauge-invariant, so this does not affect production. Worth revisiting only if a gauge-dependent quantity ever needs a gradient.

## Blast radius

Complex-valued SVD under AD, everywhere: `truncated_svd_ad`, `truncated_svd_ad_vh_only`, `regularized_svd`, and hence every complex CTM/iPEPS gradient. 751a additionally corrupts the **forward** pass, not just gradients.

Fixed together with #750; regression gates in `tests/test_svd_adjoint_fd_750.py`.

> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.
